### PR TITLE
Use model prefix also for submodels

### PR DESCRIPTION
**Pull request by stweil**
Fix also a regression in the for loop which handles submodels.

Fixes: 0d91c700c062 ("Modernize code in Tesseract::init_tesseract")
Signed-off-by: Stefan Weil <sw@weilnetz.de>

### CODE REVIEW

**Comment by stweil** [comment]
Some language models load submodels (using `tessedit_load_sublangs` in their config), for example `jpn` loads `jpn_vert`.

If such models are installed in a subdirectory (for example `best`), they can be accessed by using that subdirectory name as a prefix (`best/jpn`). Then the submodels should use the same prefix (`best/jpn_vert`) automatically.

This is achieved by this pull request.

File: src/ccmain/tessedit.cpp [hunk 1]
```diff
@@ -247,6 +249,16 @@ static bool IsStrInList(const std::string &str, const std::vector<std::string> &
 void Tesseract::ParseLanguageString(const std::string &lang_str, std::vector<std::string> *to_load,
                                     std::vector<std::string> *not_to_load) {
   std::string remains(lang_str);
+  // Look whether the model file uses a prefix which must be applied to
+  // included model files as well.
+  std::regex e("(.*)/[^/]*");
+  std::cmatch cm;
+  std::string prefix;
+  std::regex_match(lang.c_str(), cm, e, std::regex_constants::match_default);
+  if (cm.size() == 2) {
```
Comment (egorpugin) [inline src/ccmain/tessedit.cpp:258]: `if (regex_match(...`

Comment (stweil) [reply src/ccmain/tessedit.cpp]: Thanks, I changed that now.

File: src/ccmain/tessedit.cpp [hunk 2]
```diff
@@ -23,6 +23,8 @@
 #  include "config_auto.h"
 #endif
 
+#include <regex> // for std::regex_match
```
Comment (egorpugin) [inline src/ccmain/tessedit.cpp:26]: Std headers must go last.

**Pull request merged**